src/include/optimizer/rule_impls.h

```diff
@@ -0,0 +1,823 @@
+#pragma once
+
+#include "optimizer/rule.h"
+
+#include <memory>
+#include <vector>
+
+namespace terrier::optimizer {
+
+//===--------------------------------------------------------------------===//
+// Transformation rules
+//===--------------------------------------------------------------------===//
+
+/**
+ * Rule transforms (A JOIN B) -> (B JOIN A)
+ */
+class InnerJoinCommutativity : public Rule {
```

Comment:
We should be consistent in the naming for all of these rules. Each rule name should be `LogicalOp` To `Logical/Physical Op`. This one could be called `LogicalInnerJoinCommutativity`.